src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -410,29 +380,11 @@ private string GenerateForCollection(TypeGenerationSpec typeGenerationSpec)
                 TypeGenerationSpec? collectionKeyTypeMetadata = typeGenerationSpec.CollectionKeyTypeMetadata;
                 Debug.Assert(!(typeGenerationSpec.ClassType == ClassType.Dictionary && collectionKeyTypeMetadata == null));
                 string? keyTypeCompilableName = collectionKeyTypeMetadata?.TypeRef;
-                string? keyTypeReadableName = collectionKeyTypeMetadata?.TypeInfoPropertyName;
-
-                string? keyTypeMetadataPropertyName;
-                if (typeGenerationSpec.ClassType != ClassType.Dictionary)
-                {
-                    keyTypeMetadataPropertyName = "null";
-                }
-                else
-                {
-                    keyTypeMetadataPropertyName = collectionKeyTypeMetadata.ClassType == ClassType.TypeUnsupportedBySourceGen
-                        ? "null"
-                        : $"this.{keyTypeReadableName}";
```

Comment:
For my education what's the type of this member in generated code?

Reply:
so note that this is actually using JsonTypeInfo and not `Type`. It's determined by converter and if combinator is not used it will be assigned to `this.{keyTypeReadableName}`